Pilot/pilotTools.py

```diff
@@ -622,34 +639,32 @@ def __init__(self, senderFunc, bufsize=1000, autoflush=10):
             self._timer.start()
         else:
             self._timer = None
-        self.output = StringIO()
```

Comment:
There was an idea behind the buffer being a `StringIO `object. The buffer was first created in the pilot wrapper [(https://github.com/DIRACGrid/DIRAC/pull/6208/files#diff-bbbbde6222aeb320aa12b3df745daf44810996d2bf93c760e10625e1f6e4ed59)
](https://github.com/DIRACGrid/DIRAC/pull/6208/files#diff-bbbbde6222aeb320aa12b3df745daf44810996d2bf93c760e10625e1f6e4ed59) and passed to the pilot. A string appender was also added to a (classic) dirac logger). This served 2 purposes:

1. accumulate logs before the pilot was started.
2. enable to log from Dirac (not pilot) w/o any changes to Dirac code, which from now on would log to `stdout` and a string buffer.

I cannot tell at this level if your code allows remote logging from Dirac (and DiracX) using loggers they have.

Reply:
OK, so how does DiracX get hold of the buffer ? Dirac got it via a string appender to a global Logger object. Which DiracX PR is it ?